flareon/src/db.rs

```diff
@@ -621,6 +640,106 @@ impl StatementResult {
+/// database. The length is specified as a const generic parameter. The string
+/// is stored as a normal string in memory, but it is checked when it is
+/// created to ensure that it is not longer than the specified limit.
+///
+/// # Database
+///
+/// This type is represented by the `VARCHAR` type in the database, with the
+/// maximum length same as the limit specified in the type.
+///
+/// # Examples
+///
+/// ```
+/// use flareon::db::LimitedString;
+///
+/// let limited_string = LimitedString::<5>::new("test").unwrap();
+/// assert_eq!(limited_string, "test");
+///
+/// let limited_string = LimitedString::<5>::new("too long");
+/// assert!(limited_string.is_err());
+/// ```
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Deref)]
+pub struct LimitedString<const LIMIT: u32>(String);
+
+impl<const LIMIT: u32> PartialEq<&str> for LimitedString<LIMIT> {
+    fn eq(&self, other: &&str) -> bool {
+        self.0 == *other
+    }
+}
+impl<const LIMIT: u32> PartialEq<String> for LimitedString<LIMIT> {
+    fn eq(&self, other: &String) -> bool {
+        self.0 == *other
+    }
+}
+impl<const LIMIT: u32> PartialEq<LimitedString<LIMIT>> for &str {
+    fn eq(&self, other: &LimitedString<LIMIT>) -> bool {
+        *self == other.0
+    }
+}
+impl<const LIMIT: u32> PartialEq<LimitedString<LIMIT>> for String {
+    fn eq(&self, other: &LimitedString<LIMIT>) -> bool {
+        *self == other.0
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Error)]
+#[error("string is too long ({length} > {LIMIT})")]
+pub struct NewLimitedStringError<const LIMIT: u32> {
+    length: u32,
+}
+
+impl<const LIMIT: u32> LimitedString<LIMIT> {
+    pub fn new(
+        value: impl Into<String>,
+    ) -> std::result::Result<Self, NewLimitedStringError<LIMIT>> {
+        let value = value.into();
+        let length = value.len() as u32;
+
+        if length > LIMIT {
+            return Err(NewLimitedStringError { length });
+        }
+        Ok(Self(value))
+    }
+}
+
+#[cfg(feature = "fake")]
+impl<const LIMIT: u32> fake::Dummy<usize> for LimitedString<LIMIT> {
+    fn dummy_with_rng<R: rand::Rng + ?Sized>(len: &usize, rng: &mut R) -> Self {
+        use rand::Rng;
+
+        assert!(
+            *len <= LIMIT as usize,
+            concat!(
+                "len must be less than or equal to LIMIT (",
+                stringify!(LIMIT),
+                ")"
+            )
+        );
+
+        let str: String = rng
+            .sample_iter(&rand::distributions::Alphanumeric)
+            .take(*len)
+            .map(char::from)
+            .collect();
+        Self::new(str).unwrap()
+    }
+}
+
+#[cfg(feature = "fake")]
+impl<const LIMIT: u32> fake::Dummy<fake::Faker> for LimitedString<LIMIT> {
+    fn dummy_with_rng<R: rand::Rng + ?Sized>(_: &fake::Faker, rng: &mut R) -> Self {
+        use fake::Fake;
+
+        let len: usize = (0..LIMIT as usize).fake_with_rng(rng);
+        len.fake_with_rng(rng)
+    }
+}
```

Comment:
I don't see it being used anywhere

Reply:
It's being used here: https://github.com/flareon-rs/flareon/blob/ebf504ef5b48c091b80d79d85819d5eec95f4d7a/flareon/tests/db.rs#L170

Reply:
Missed that, that's fine then!